src/lib/InstantSearch.ts

```diff
@@ -285,10 +314,11 @@ See: https://www.algolia.com/doc/guides/building-search-ui/going-further/backend
 
     this.mainHelper = mainHelper;
 
-    this.mainIndex.init({
+    // @TODO: do we want this to be correct arguments?
+    this.mainIndex.init(({
       instantSearchInstance: this,
       parent: null,
-    });
+    } as any) as InitOptions);
```

Comment:
we could change the signature of Index to only need these options

Reply:
Since we correctly type our widget lifecycle now, it would make sense to either create a 
`IndexInitOptions` type or to pass all options.

I would go for passing all init options because we might need them at some point in the `index` methods and the type checking might break if we keep casting.

Reply:
`IndexInitOptions` seems a good compromise, the type should be a subset of `InitOptions` anyway.